service/history/replicationTaskProcessor.go

```diff
@@ -153,6 +151,7 @@ Loop:
 		// for each iteration, do close check first
 		select {
 		case <-p.done:
+			p.logger.Info("ReplicationTaskProcessor shutting down.")
```

Comment:
can you separate the logic into dedicated functions?

Reply:
done